pkg/providers/snow/apibuilder.go

```diff
@@ -135,11 +137,46 @@ func SnowCluster(clusterSpec *cluster.Spec) *snowv1.AWSSnowCluster {
 				Host: clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host,
 				Port: 6443,
 			},
+			IdentityRef: &snowv1.AWSSnowIdentityReference{
+				Name: CredentialsSecretName(clusterSpec),
+				Kind: snowv1.SecretKind,
+			},
 		},
 	}
 	return cluster
 }
 
+func CredentialsSecret(name, namespace, credsB64, certsB64 string) *v1.Secret {
+	return &v1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1.SchemeGroupVersion.String(),
+			Kind:       string(snowv1.SecretKind),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			v1alpha1.SnowCredentialsKey:  []byte(credsB64),
+			v1alpha1.SnowCertificatesKey: []byte(certsB64),
+		},
+		Type: v1.SecretTypeOpaque,
+	}
+}
+
+func SnowCredentialsSecret(clusterSpec *cluster.Spec, credsB64, certsB64 string) *v1.Secret {
```

Comment:
nit: to make the distinction clear, it's not for snow in general, it's for the snow capi provider

```suggestion
func CAPASCredentialsSecret(clusterSpec *cluster.Spec, credsB64, certsB64 string) *v1.Secret {
```